internal/host/plugin/repository_host_catalog.go

```diff
@@ -59,23 +62,38 @@ func (r *Repository) CreateCatalog(ctx context.Context, c *HostCatalog, _ ...Opt
 	}
 	c.PublicId = id
 
-	oplogWrapper, err := r.kms.GetWrapper(ctx, c.ScopeId, kms.KeyPurposeOplog)
+	plgClient, ok := r.plugins[plg.GetPublicId()]
+	if !ok {
+		return nil, errors.New(ctx, errors.Internal, op, fmt.Sprintf("expected plugin %q not available", plg.GetPluginName()))
+	}
+	plgHc, err := toPluginCatalog(ctx, c)
 	if err != nil {
-		return nil, errors.Wrap(ctx, err, op, errors.WithMsg("unable to get oplog wrapper"))
+		return nil, errors.Wrap(ctx, err, op)
 	}
-	dbWrapper, err := r.kms.GetWrapper(ctx, c.ScopeId, kms.KeyPurposeDatabase)
+	plgResp, err := plgClient.OnCreateCatalog(ctx, &plgpb.OnCreateCatalogRequest{Catalog: plgHc})
 	if err != nil {
-		return nil, errors.Wrap(ctx, err, op, errors.WithMsg("unable to get db wrapper"))
+		return nil, errors.Wrap(ctx, err, op)
```

Comment:
The error should be ignored if it's an unimplemented error. There's no particular reason that plugins _must_ include an OnCreateCatalog function; if they don't the Unimplemented one will return an error that we can safely ignore.

As a corollary this means there needs to be a nil check on `plgResp` before using it in code below.

Reply:
We no longer error on unimplemented.  Added tests for this as well.